include/node.tcc

```diff
@@ -346,6 +353,57 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_velocity_constraints() {
   }
 }
 
+//My Idea for what to input
+
+// bool assign_nodal_absorbing_constraints(double pwave_v, double swave_v);
+// !Assign absorbing constraints
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::assign_nodal_absorbing_constraint(
+    double pwave_v, double swave_v) {
+  bool status = true;
+  // Assign shear velocity constraint
+  try{
+  	unsigned dir0= 0;
+    const auto s_phase = static_cast<unsigned>(dir0/Tdim);
+    double s_mass_density = mass_(s_phase)/volume_(s_phase);
+    double s_velocity = velocity_(dir0,s_phase);
+    double s_traction = -s_velocity*s_mass_density*swave_v;
+
+  	unsigned dir1 = 1;
+  	const auto c_phase = static_cast<unsigned>(dir1/Tdim);
+  	double n_mass_density = mass_(c_phase)/volume_(c_phase);
+  	double n_velocity = velocity_(dir1,c_phase);
+    double n_traction = -n_velocity*n_mass_density*pwave_v;
+
+  	this->absorbing_constraints_.insert(std::make_pair<unsigned,double>(
+  		static_cast<unsigned>(dir0), static_cast<double>(s_traction)));
+  		
+
+    this->absorbing_constraints_.insert(std::make_pair<unsigned,double>(
+  		static_cast<unsigned>(dir1), static_cast<double>(n_traction)));
+
+  } catch (std::exception& exception) {
+    console_->error("{} #{}: {}\n", __FILE__, __LINE__, exception.what());
+    status = false;
+  }
+  return status;
+}
+
+// !Apply absorbing constraints
```

Comment:
I think you get `assign` and `apply` the other way around. I think this one is for `assign` and the other one, the real computation, is for `apply`